docs/extensions/api.md

```diff
@@ -3,24 +3,31 @@ title:      Extensions API
 # Writing Extensions for Python-Markdown
 
 Python-Markdown includes an API for extension writers to plug their own
-custom functionality and/or syntax into the parser. There are Preprocessors
-which allow you to alter the source before it is passed to the parser,
-inline patterns which allow you to add, remove or override the syntax of
-any inline elements, and Postprocessors which allow munging of the
-output of the parser before it is returned. If you really want to dive in,
-there are also Blockprocessors which are part of the core BlockParser.
-
-As the parser builds an [ElementTree][ElementTree] object which is later rendered
-as Unicode text, there are also some helpers provided to ease manipulation of
-the tree. Each part of the API is discussed in its respective section below.
-Additionally, reading the source of some [Available Extensions][] may be
-helpful. For example, the [Footnotes][] extension uses most of the features
-documented here.
+custom functionality and syntax into the parser. An extension will patch
+into one or more stages of the parser:
+
+* [*Preprocessors*](#preprocessors) alter the source before it is passed to the parser. 
+* [*Block Processors*](#blockprocessors) work with blocks of text separated by blank lines.
+* [*Inline Patterns*](#inlinepatterns) work with inline elements, such as `*strong*`. 
+* [*Tree Processors*](#treeprocessors) modify the constructed ElementTree
+* [*Postprocessors*](#postprocessors) munge of the output of the parser just before it is returned. 
+
+The parser loads text, applies the preprocessors, creates an
+[ElementTree][ElementTree] added to by the block processors and 
+inline patterns, renders the ElementTree as Unicode text, and then 
+then applies the postprocessors.
+
+There are classes and helpers provided to ease writing your extension. 
+Each part of the API is discussed in its respective section below.
+Additionally, you can walk through the [Tutorial on Writing Extensions][tutorial];
```

Comment:
The tutorial is out-of-date. It was never updated for version 3.0 and won't work as-is. While I agree that linking to the tutorial here would be ideal, I think the tutorial needs to be updated first.

Reply:
The tutorial is up-to-date.   I updated it.

Reply:
awesome. Thanks